src/mongoose_wpool.erl

```diff
@@ -9,35 +9,135 @@
 -author("bartlomiej.gorny@erlang-solutions.com").
 -include("mongoose.hrl").
 -include("mongoose_wpool.hrl").
+
+-record(mongoose_wpool, {name :: term(), strategy :: atom(), call_timeout :: pos_integer()}).
+
 %% API
--export([ensure_started/0, get_pool_settings/1, delete_pool_settings/1, save_pool_settings/2]).
+-export([ensure_started/0,
+         start/2, start/3, start/4,
+         stop/1, stop/2, stop/3,
+         get_worker/1, get_worker/2, get_worker/3,
+         call/2, call/3, call/4, call/5,
+         cast/2, cast/3, cast/4, cast/5,
+         get_pool_settings/3, get_pools/0, stats/3]).
 
 ensure_started() ->
     wpool:start(),
-    case ets:info(pool_settings_tab()) of
+    case ets:info(?MODULE) of
         undefined ->
             % we set heir here because the whole thing may be started by an ephemeral process
-            ets:new(pool_settings_tab(), [named_table, public,
+            ets:new(?MODULE, [named_table, public,
+                {read_concurrency, true},
                 {keypos, 2},
                 {heir, whereis(wpool_sup), undefined}]);
         _ ->
             ok
     end.
 
-get_pool_settings(PoolName) ->
-    case ets:lookup(pool_settings_tab(), PoolName) of
-        [PoolOpts] ->
-            PoolOpts;
+start(Type, PoolOpts) ->
+    start(Type, global, PoolOpts).
+
+start(Type, Host, PoolOpts) ->
+    start(Type, Host, default, PoolOpts).
+
+start(Type, Host, Tag, PoolOpts) ->
+    {Opts0, WpoolOpts} = proplists:split(PoolOpts, [strategy, call_timeout]),
+    Opts = lists:append(Opts0),
+    case wpool:start_sup_pool(make_pool_name(Type, Host, Tag), WpoolOpts) of
+        {ok, Pid} ->
+            Strategy = proplists:get_value(strategy, Opts, best_worker),
```

Comment:
It looks to me that the `strategy` option will land in `WpoolOpts` var as in line [44](https://github.com/esl/MongooseIM/pull/2060/files#diff-293b45de04be8a714405e0a6abdd3048R44). This means that `Opts` var will never have `strategy` option and this also means that the `Strategy` var in this line is always `best_worker`.

What did I miss in my reasoning?

Reply:
Ah, sorry may bad... It works as expected, ignore my previous comment.